Minesweeper/Minesweeper/Entities/MineField.swift

```diff
@@ -0,0 +1,116 @@
+//
+//  MineField.swift
+//  Minesweeper
+//
+//  Created by Dorian Grolaux on 19/02/2018.
+//
+
+import Foundation
+
+typealias Coordinates = (x: Int, y: Int)
+
+class MineField {
+    private let cells: [[Cell]]!
+    
+    init(cells: [[Cell]]) {
+        self.cells = cells
+    }
+    
+    
+    init(gridSize size: Int, withBombAmount bombs: Int) {
+        
+        let randomPlacements = MineField.getPlacementOptions(count: bombs, forArraySize: size)
+        var cells = [[Cell]]()
+        
+        for x in 0..<size {
+            var yCells = [Cell]()
+            
+            for y in 0..<size {
+                yCells.append(Cell(containsBomb: randomPlacements.contains(where: { placement -> Bool in
+                    return placement == (x: x, y: y)
+                })))
+            }
+            cells.append(yCells)
+        }
+        
+        MineField.markPlacedBombs(inField: &cells, placedBombs: randomPlacements)
+        
+        self.cells = cells
+    }
+    
```

Comment:
This is a playground I made with an idea of how we'd use subscript here. Just replace the Ints I used to test it with your `Cell` objects.
```
struct Grid {
    private let grid = [[1,2,3],[4,5,6],[7,8,9]]
    
    subscript(x: Int, y: Int) -> Int? {
        guard let firstRow = grid.first, let firstCell = firstRow.first else { return nil}
        let totalRows = grid.count
        let totalColumns = firstRow.count
        guard x < totalColumns && y < totalRows else { return nil }
        return grid[y][x]
    }
}

let grid = Grid()

grid[1,2]
grid[0,0]
grid[1,0]

print("""
[1] [2] [3]
[4] [5] [6]
[7] [8] [9]
""")


//Better than Grid.grid[x][y] and returns nil when out of bounds
```

Reply:
Only thing I'm not sure on is returning optionals when the index is out of bounds, I think throwing an error or using `Assert` to check these values is better as we don't really want to provide optionals for this but immediately warn ourselves we've gone and done this as the user shouldn't be able to reach such a state.

Reply:
Done